runelite-client/src/main/java/net/runelite/client/plugins/pyramidplunder/PyramidPlunderPlugin.java

```diff
@@ -0,0 +1,204 @@
+import javax.inject.Inject;
+import lombok.Getter;
+import net.runelite.api.Client;
+import net.runelite.api.Constants;
+import net.runelite.api.GameObject;
+import net.runelite.api.GameState;
+import static net.runelite.api.ItemID.PHARAOHS_SCEPTRE;
+import net.runelite.api.Tile;
+import net.runelite.api.TileObject;
+import net.runelite.api.Varbits;
+import net.runelite.api.WallObject;
+import net.runelite.api.events.GameObjectSpawned;
+import net.runelite.api.events.GameStateChanged;
+import net.runelite.api.events.GameTick;
+import net.runelite.api.events.WallObjectSpawned;
+import net.runelite.api.widgets.Widget;
+import net.runelite.api.widgets.WidgetInfo;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.eventbus.Subscribe;
+import net.runelite.client.game.ItemManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import static net.runelite.client.plugins.pyramidplunder.PyramidPlunderObjects.GRAND_GOLD_CHEST_ID;
+import static net.runelite.client.plugins.pyramidplunder.PyramidPlunderObjects.SARCOPHAGUS_ID;
+import static net.runelite.client.plugins.pyramidplunder.PyramidPlunderObjects.SPEARTRAP_ID;
+import static net.runelite.client.plugins.pyramidplunder.PyramidPlunderObjects.TOMB_DOOR_WALL_IDS;
+import static net.runelite.client.plugins.pyramidplunder.PyramidPlunderObjects.URN_IDS;
+import net.runelite.client.ui.overlay.OverlayManager;
+import net.runelite.client.ui.overlay.infobox.InfoBoxManager;
+
+@PluginDescriptor(
+	name = "Pyramid Plunder",
+	description = "Show custom overlay for Pyramid Plunder",
+	tags = {"minigame"},
+	enabledByDefault = false
+)
+public class PyramidPlunderPlugin extends Plugin
+{
+	// Total number of game ticks until the pyramid plunder round ends (5 minutes)
+	private static final int PYRAMID_PLUNDER_DURATION = 500;
+	private static final int PYRAMID_PLUNDER_REGION = 7749;
+
+	@Inject
+	private Client client;
+
+	@Inject
+	private PyramidPlunderConfig config;
+
+	@Inject
+	private OverlayManager overlayManager;
+
+	@Inject
+	private PyramidPlunderOverlay overlay;
+
+	@Inject
+	private InfoBoxManager infoBoxManager;
+
+	@Inject
+	private ItemManager itemManager;
+
+	@Getter
+	private final Map<TileObject, Tile> tilesToHighlight = new HashMap<>();
+
+	@Getter
+	private final List<GameObject> objectsToHighlight = new ArrayList<>();
+
+	private PyramidPlunderTimer timer;
+
+	@Provides
+	PyramidPlunderConfig getConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(PyramidPlunderConfig.class);
+	}
+
+	@Override
+	protected void startUp() throws Exception
+	{
+		overlayManager.add(overlay);
+	}
+
+	@Override
+	protected void shutDown() throws Exception
+	{
+		tilesToHighlight.clear();
+		objectsToHighlight.clear();
+		overlayManager.remove(overlay);
+		timer = null;
+		infoBoxManager.removeIf(infoBox -> infoBox instanceof PyramidPlunderTimer);
+
+		Widget ppWidget = client.getWidget(WidgetInfo.PYRAMID_PLUNDER_DATA);
```

Comment:
this should be in an `clientThread.invoke()`

Reply:
Just for my understanding, why should this be in `clientThread.invoke()`?

Reply:
shutDown runs on the EDT and it is mostly not safe to mutate client state off of the client thread.